src/query/api/v1/handler/placement/common.go

```diff
@@ -130,3 +143,21 @@ func RegisterRoutes(r *mux.Router, client clusterclient.Client, cfg config.Confi
 	r.HandleFunc(AddURL, logged(NewAddHandler(client, cfg)).ServeHTTP).Methods(AddHTTPMethod)
 	r.HandleFunc(DeleteURL, logged(NewDeleteHandler(client, cfg)).ServeHTTP).Methods(DeleteHTTPMethod)
 }
+
+func validateAllAvailable(p placement.Placement) error {
+	badInsts := []string{}
+	for _, inst := range p.Instances() {
+		if inst.Shards().NumShards() == 0 {
+			continue
```

Comment:
I don't think this is possible in valid placement? could you add this to the list of `badInsts` for now and return as error?

Reply:
Hmm if it's not possible in a valid placement I'll revert this. I mainly did it because I had a test with an instance with no shards

Reply:
Yeah we are checking against instance with no shards in a sharded placement, so this should not be possible